pkg/cloud/services/secretsmanager/cloudinit.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//go:generate go run ../../../../hack/tools/embed-file-constant/main.go secretsmanager secret-fetch.sh secretFetchScript zz_generated.secret-fetch.go
+
+package secretsmanager
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"mime/multipart"
+	"net/textproto"
+	"strings"
+
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/scope"
+)
+
+const (
+	includePart = "file:///etc/secret-userdata.txt\n"
+)
+
+var (
+	includeType = textproto.MIMEHeader{
+		"content-type": {"text/x-include-url"},
+	}
+
+	boothookType = textproto.MIMEHeader{
+		"content-type": {"text/cloud-boothook"},
+	}
+
+	multipartHeader = strings.Join([]string{
+		"MIME-Version: 1.0",
+		"Content-Type: multipart/mixed; boundary=\"%s\"",
+		"",
+	}, "\n")
+
+	secretFetchTemplate = template.Must(template.New("secret-fetch-script").Parse(secretFetchScript))
+)
+
+type scriptVariables struct {
+	SecretID string
+	Region   string
+}
+
+func (s *Service) EncryptedCloudInit(m *scope.MachineScope) (string, error) {
```

Comment:
```suggestion
func (s *Service) GetEncryptedBootstrapData(m *scope.MachineScope) (string, error) {
```

or something similar